Extension/src/LanguageServer/cppbuildTaskProvider.ts

```diff
@@ -0,0 +1,320 @@
+        const isCl: boolean = compilerPathBase === "cl.exe";
+        const isWindows: boolean = os.platform() === 'win32';
+        const cwd: string = isCl ? "${workspaceFolder}" : path.dirname(compilerPath);
+        let args: string[] = isCl ? ['/Zi', '/EHsc', '/Fe:', filePath + '.exe', '${file}'] : ['-g', '${file}', '-o', filePath + (isWindows ? '.exe' : '')];
+        if (definition === undefined && compilerArgs && compilerArgs.length > 0) {
+            args = args.concat(compilerArgs);
+        }
+        const options: cp.ProcessEnvOptions | undefined = {"cwd": cwd};
+
+        // Double-quote the command if it is not already double-quoted.
+        let resolvedcompilerPath: string = isCl ? compilerPathBase : compilerPath;
+        if (resolvedcompilerPath && !resolvedcompilerPath.startsWith("\"") && resolvedcompilerPath.includes(" ")) {
+            resolvedcompilerPath = "\"" + resolvedcompilerPath + "\"";
+        }
+
+        if (definition === undefined) {
+            definition = {
+                type: CppBuildTaskProvider.CppBuildScriptType,
+                label: taskName,
+                command: resolvedcompilerPath,
+                args: args,
+                options: options
+            };
+        }
+
+        if (returnCompilerPath) {
+            definition = definition as CppBuildTaskDefinition;
+            definition.compilerPath = isCl ? compilerPathBase : compilerPath;
+        }
+
+        let activeClient: Client = ext.getActiveClient();
+        let uri: vscode.Uri | undefined = activeClient.RootUri;
+        if (!uri) {
+            throw new Error("No client URI found in getBuildTasks()");
+        }
+        const target: vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(uri);
+        if (!target) {
+            throw new Error("No target WorkspaceFolder found in getBuildTasks()");
+        }
+
+        let task: vscode.Task =  new vscode.Task(definition, target, taskName, CppBuildTaskProvider.CppBuildSourceStr,
+            new vscode.CustomExecution(async (): Promise<vscode.Pseudoterminal> =>
+            // When the task is executed, this callback will run. Here, we setup for running the task.
+			 new CustomBuildTaskTerminal(resolvedcompilerPath, args, options, target.name)
+            ), isCl ? '$msCompile' : '$gcc');
+
+        /* const normalcommand: vscode.ShellExecution = new vscode.ShellExecution(compilerPath, [...args], { cwd: cwd });
+        let task: vscode.Task = new vscode.Task(definition, target, taskName, CppBuildTaskProvider.CppBuildSourceStr,
+            normalcommand, isCl ? '$msCompile' : '$gcc');*/
+        task.group = vscode.TaskGroup.Build;
+
+        return task;
+    };
+}
+
+class CustomBuildTaskTerminal implements vscode.Pseudoterminal {
+    private writeEmitter  = new vscode.EventEmitter<string>();
+    // eslint-disable-next-line no-invalid-this
+    onDidWrite: vscode.Event<string> = this.writeEmitter.event;
```

Comment:
Perhaps implement these are properties, so we don't have to disable lint errors for them?  i.e. public get onDidWrite(): vscode.Event<string> { return this.writeEmitter.event; }